### DISCUSSION TITLE
Ability to fill a habit in details page

### DISCUSSION THREAD

**Discussion by tasomaniac** [iSoron/uhabits#1121] [Feature Requests]
Details page does not have an option to mark a habit done. I'm missing this quite a bit actually. Would you be open for a contribution?

I'm thinking it can be a row at the top below the toolbar. It can also make use of the optional question that is displayed in notifications.

**Comment by hiqua** [comment]
Can you explain why you miss this feature? I think that wouldn't hurt to add this possibility.

**Reply by tasomaniac** [reply]
I simply open it and while browsing I remember I need to fill it for today. It is actually the same when I click a notification. In both cases, I need to go back to the main list

**Reply by mdgarf** [reply]
I agree 
Only last notification has option done/not done, if you click on later notification it will get you to habit detail and you can not "done / not done" there you have to back track to main page. 

So either from notification to main page or add possibility to enter "done / not done" in the detail page. 

Or is there a flaw how I use the app?

**Reply by mdgarf** [reply]
almost forgot 
nice app, really well done, clean and simple, no permissions ...
thanks

**Comment by LordJon** [comment]
You can edit the calendar to add it as done. But I see why having a button would require less steps.
